### PR TITLE
Homework12: Mariya Simonenko

### PR DESCRIPTION
**Pull request by Mary-si**

### CODE REVIEW

**Comment by Stml89** [comment]
You still have 4 files in a conflict state, this should be fixed. What is more, a lot of files from previous PRs, so your branch is not based on `master`\`main` branch

File: homework12_cards.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        self.mast = mast
+
+    def __repr__(self):
+        return f"{self.number} {self.mast}" if self.mast else f"{self.number}"
+
+
+class CardsDeck:
+    def __init__(self):
+        self.cards = [Card(number, mast) for number in Card.number_list[:-1] for mast in Card.mast_list]
+        self.cards.extend([Card("Joker", None) for _ in range(2)])  # Добавляем двух джокеров
```
Comment (Stml89) [inline homework12_cards.py:27]:
```suggestion
        self.cards.extend([Card("Joker", "Red")])
        self.cards.extend([Card("Joker", "Black")])
```

File: homework12_cards.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+class Card:
+    number_list = [2, 3, 4, 5, 6, 7, 8, 9, "J", "Q", "K", "A", "Joker"]
+    mast_list = ["clubs", "diamonds", "hearts", "spades"]
+
+    def __init__(self, number, mast):
+        self.number = number
+        self.mast = mast
+
+    def __repr__(self):
+        return f"{self.number} {self.mast}" if self.mast else f"{self.number}"
```
Comment (Stml89) [inline homework12_cards.py:21]:
```suggestion
        return f"{self.number} {self.mast}"
```

File: homework12_converter.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+vasya = Currency("USD", 10)
+petya = Currency("EUR", 5)
+
+# Если валюта не задана, то конвертация происходит в BYN:
+assert Currency.exchange_currency(vasya.currency, vasya.amount) == (32.69, "BYN"), "Ошибка конвертации в BYN"
+assert Currency.exchange_currency(petya.currency, petya.amount) == (35.20, "BYN"), "Ошибка конвертации в BYN"
+
+# Конвертация в заданную валюту:
+assert Currency.exchange_currency(vasya.currency, vasya.amount) == (9.29, "EUR"), "Ошибка конвертации в EUR"
+assert Currency.exchange_currency(petya.currency, petya.amount) == (10.76, "USD"), "Ошибка конвертации в USD"
```
Comment (Stml89) [inline homework12_converter.py:92]: have you tried to run this program, it does not work =(

File: homework9_candles.py
Comment (Stml89) [inline homework9_candles.py:1]: File(s) related to prev HW

File: homework12_molecules.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+# Mg(OH)2 -> {Mg: 1, O: 2, H: 2}
+# K4[ON(SO3)2]2 -> {K: 4, O: 14, N: 2, S: 4}
+
+
+import re
+from collections import Counter
+
+
+def repl(m):
+    """возвращает строко, ктр состоит из символов элемента"""
```
Comment (Stml89) [inline homework12_molecules.py:23]: typos in doc string

File: homework12_molecules.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    return m[1] * int(m[2] if m[2] else 1)
+
+
+def molecule(formula: str) -> dict:
```
Comment (Stml89) [inline homework12_molecules.py:27]: Brackets also can be `{` `}`

**Review by Stml89 — CHANGES_REQUESTED** [review]

**Comment by Mary-si** [comment]
+

**Comment by Stml89** [comment]
1\ add `.idea` to the `.gitignore` file. 
2\ Fix comments above

**Review by Stml89 — CHANGES_REQUESTED** [review]
Comment above

**Comment by Mary-si** [comment]
отредактировала

**Comment by Mary-si** [comment]
тут ок?

**Comment by Stml89** [comment]
как минимум надо удалить папку `.idea` из репозитория

**Comment by Mary-si** [comment]
проверь пожалуйста

**Review by Stml89 — CHANGES_REQUESTED** [review]
Overall, HW looks good, but you need to fix the comments and remove the `.idea` directory. I marked it with 6, but when you fix the comments, I'll revise the grade.

**Comment by Mary-si** [comment]
еще раз глянь

**Review by Stml89 — APPROVED** [review]

**Pull request merged**